### ISSUE TITLE
With TypeScript, stack traces are off

### ISSUE THREAD

**Issue by krzkaczor** [nomiclabs/hardhat#1336]
When using hardhat with TS regular error stack traces (coming from js not sol) are off. 

Repro: https://github.com/krzkaczor/typescript-solidity-dev-starter-kit/tree/kk/repro-stack-traces

Basically error is:
`      at ...typescript-solidity-dev-starter-kit/test/counter.ts:54:31`
but it should be line 14.

I tried playing with tsconfig (enabling inline sourcemaps etc, bumping target). Stack traces seems to be fine in `hardhatconfig.ts` - they are only wrong in tests.

Any ideas? I can work on this but I am not that familiar with codebase so any pointers would be great.

**Comment by krzkaczor** [comment]
I tracked it down to issue with waffle. Closing here.

**Comment by krzkaczor** [comment]
I figured it out. This bug is triggered if you import anything from waffle directly in your tests. 

I used rhlsthrm/typescript-solidity-dev-starter-kit boilerplate which had this bug. 

Here's a PR to fix this in dev-starter-kit: https://github.com/rhlsthrm/typescript-solidity-dev-starter-kit/pull/19

and here's a PR to lazy load ganache in waffle to avoid triggering this bug: https://github.com/EthWorks/Waffle/pull/471
